lib/active_merchant/billing/gateways/braintree_blue.rb

```diff
@@ -304,9 +304,9 @@ def merge_credit_card_options(parameters, options)
 
         parameters[:credit_card] ||= {}
         parameters[:credit_card].merge!(:options => valid_options)
-        address = options[:billing_address]&.except(:phone)
-        return parameters if address.nil? || address.values.compact.empty?
-        parameters[:credit_card][:billing_address] = map_address(address)
+        billing_address = map_address(options[:billing_address]) if options[:billing_address]
+        return parameters if billing_address.nil? || billing_address.values.compact.empty?
```

Comment:
Okay, if we're going back in here anyway, let's fix this to `billing_address.all? { |_k, v| v.nil? || v.empty? }`